misc.go

```diff
@@ -338,6 +346,42 @@ func max(a, b int) int {
 	return b
 }
 
+// This function parses given locale string into language tag value. Passing empty
+// string as locale means reading locale value from environment.
+func getLocaleTag(localeStr string) (language.Tag, error) {
+	localeTag := language.Und
+	var err error
+
+	if localeStr == localeStrSys {
+		// read environment locale
+		return locale.Detect()
+	}
+
+	localeTag, err = language.Parse(localeStr)
```

Comment:
NIT: You can remove the `var err error` declaration above:

```suggestion
	localeTag, err := language.Parse(localeStr)
```